src/main/java/org/openstreetmap/atlas/checks/validation/relations/DuplicateRelationCheck.java

```diff
@@ -0,0 +1,143 @@
+package org.openstreetmap.atlas.checks.validation.relations;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMember;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMemberList;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * This check looks for two or more {@link Relation}s duplicate with same OSM tags and same members
+ * with same roles.
+ *
+ * @author Xiaohong Tang
+ */
+public class DuplicateRelationCheck extends BaseCheck<Object>
+{
+    public static final String DUPLICATE_RELATION_INSTRUCTIONS = "Relation {0} and {1} are duplicates with same OSM tags and same members with same roles.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList(DUPLICATE_RELATION_INSTRUCTIONS);
+    private static final long serialVersionUID = 2723186269280026809L;
+
+    public DuplicateRelationCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Relation && !isFlagged(object.getOsmIdentifier());
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Relation obj = (Relation) object;
```

Comment:
Please don't use abbreviations.
```suggestion
        final Relation relation = (Relation) object;
```

Reply:
thanks, updated it now